### PR TITLE
Email auth

### LINKED ISSUE #7: Change user_name auth to email auth.
- [ ] Auth via email
- [ ] Implement token authentication (TokenAuthentication)

### PR DESCRIPTION
**Pull request by EdiOancea**
Started from #12 
Closes #7

### CODE REVIEW

File: app/urls.py [hunk 1]
```diff
@@ -1,9 +1,12 @@
-from django.conf import settings
+from django.conf.urls import url
```
Comment (raresluncan) [inline app/urls.py]: Look into 'path' instead of url (found in django 2.0 + docs).

File: app/views/user.py [hunk 2]
```diff
@@ -1,10 +1,41 @@
...
+        user = get_object_or_404(queryset, pk=pk)
+        user.undelete()
+        serializer = UserSerializer(user)
+
+        return Response(serializer.data)
+
+    def update(self, request, pk=None):
+        queryset = User.objects.all()
+        user = get_object_or_404(queryset, pk=pk)
+        existingUser = UserSerializer(user)
```
Comment (raresluncan) [inline app/views/user.py]: We should not use camel case for python variables (except class names).

File: app/views/user.py [hunk 3]
```diff
@@ -1,10 +1,41 @@
...
+
+        return super(UserViewSet, self).update(request, pk)
+
+    def create(self, request):
+        createdUser = User.objects.create_user(**request.data)
```
Comment (raresluncan) [inline app/views/user.py]: !camelCase.

File: app/views/user.py [hunk 4]
```diff
@@ -1,10 +1,41 @@
...
 from rest_framework.response import Response
+from rest_framework.decorators import action
 
 from app.models.user import User
 from app.serializers.user import UserSerializer
 
 class UserViewSet(viewsets.ModelViewSet):
     serializer_class = UserSerializer
     queryset = User.objects.all()
+    permission_classes = (permissions.AllowAny,)
```
Comment (raresluncan) [inline app/views/user.py:12]: I think we shouldn't allow any for user operations.

File: app/views/user.py [hunk 5]
```diff
@@ -1,10 +1,41 @@
...
+    @action(detail=True, methods=['put'])
+    def reactivate(self, request, pk=None):
+        queryset = User.objects.get_all()
+        user = get_object_or_404(queryset, pk=pk)
+        user.undelete()
+        serializer = UserSerializer(user)
+
+        return Response(serializer.data)
+
+    def update(self, request, pk=None):
```
Comment (raresluncan) [inline app/views/user.py]: We can remove this.

File: app/views/user.py (hunk 2 above)
Comment (EdiOancea) [reply app/views/user.py]: :(

**Pull request merged**